generate_uninstall/CMakeLists.txt

```diff
@@ -3,6 +3,14 @@
 
 set (UNINSTALL_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 
+# When added as a subdirectory in the mongocxx driver, there is a conflict in
+# the `uninstall` target causing the `make uninstall` command to fail. By
+# defining the uninstall target as a variable, the C++ driver is able to rename
+# this target in order to avoid the conflict.
+if (NOT DEFINED MONGOC_UNINSTALL_TARGET)
```

Comment:
Per https://gitlab.kitware.com/cmake/cmake/-/issues/23024#note_1098749, it may actually be more reasonable to simply not define the uninstall target at all if this project is not the top-level project (note: `PROJECT_IS_TOP_LEVEL` is a CMake 3.21 feature). The top-level project (i.e. the CXX Driver) would be responsible for defining the `uninstall` target behavior to include (or ignore) this project's files.
```cmake
if ("${CMAKE_SOURCE_DIR}" STREQUALS "${PROJECT_SOURCE_DIR}")
  add_custom_target(uninstall ...)
endif ()
```

Reply:
This seems to work too. I don't have an opinion on which one we go with.